src/util/snapshots.tsx

```diff
@@ -0,0 +1,78 @@
+import { LxdInstance } from "types/instance";
+import { TestFunction } from "yup";
+import { AnyObject } from "yup/lib/types";
+import { checkDuplicateName, getTomorrow } from "./helpers";
+
+export interface SnapshotFormValues {
+  name: string;
+  stateful: boolean;
+  expirationDate: string | null;
+  expirationTime: string | null;
+}
+
+export const getExpiresAt = (
+  expirationDate: string,
+  expirationTime: string | null
+) => {
+  expirationTime = expirationTime ?? "00:00";
+  return `${expirationDate}T${expirationTime}`;
+};
+
+export const testDuplicateName = (
+  instance: LxdInstance,
+  controllerState: [
+    AbortController | null,
+    React.Dispatch<React.SetStateAction<AbortController | null>>
+  ],
+  excludeName?: string
+): [string, string, TestFunction<string | undefined, AnyObject>] => {
+  return [
+    "deduplicate",
+    "Snapshot name already in use",
+    (value?: string) => {
+      return (
+        (excludeName && value === excludeName) ||
+        checkDuplicateName(
+          value,
+          instance.project,
+          controllerState,
+          `instances/${instance.name}/snapshots`
+        )
+      );
+    },
+  ];
+};
+
+export const testForbiddenChars = (): [
```

Comment:
A whitelist might be better, I think we only allow a-z, 0-9 and -.

Reply:
A place where I could check this would help. The `.` is also allowed, so I need to know exactly what characters to allow in order to make a whitelist.

Reply:
All I found was [this list](https://github.com/lxc/lxd/blob/086f33b9b0f198eb91e551a75d00d1fc9bfe5650/lxd/instance/instance_utils.go#L705), which is exactly like yours. When testing also other special chars were rejected, so there must be other places validating the name.

Reply:
Yeah, I also tested with different characters yesterday. What I noticed is that the error messages are often different, meaning that they're probably validated in different places. I propose to note this down for a future improvement, and check with the core team if they can provide us with an exhaustive list of allowed characters.